frontend/src/page/profile.tsx

```diff
@@ -27,6 +29,22 @@ export const Profile: React.FC = () => {
     cancel(profileUpdateFetch);
   }, []);
 
+  React.useEffect(() => {
+    if (userAvatar) {
+      const avatarCreatePromise = abortableJsonFetch<StoredUserAvatar>(
+        API_PROFILE_AVATAR_CREATE,
+        Auth.Required,
+        {},
+        {
+          body: userAvatar.file,
+          method: "POST"
+        }
+      );
+      (async () => {
+        await avatarCreatePromise;
```

Comment:
Hmm now that I'm using the user id as the id for the avatar, I'm running into a small problem because the avatar won't update on the client when a new one is uploaded.

Do you think we could put some kind of uuid in the scheme for avatars?
`/users/avatar/:user_id/:uuid`

Otherwise I guess there are probably ways in React to "force" a re-render, but it seems like we would want to avoid that if possible.

Let me know if any other ideas come to mind. :) @TrevorSundberg

Reply:
Yeah I've definitely run into that sort of problem. I think the solution is to:
```
await avatarCreatePromise;
setUser({...user});
```

Which is equivalent to forcing a re-render, however it's at least a bit more proper because it technically only updates things that depend on user (though that's probably everything in this case). I believe that's an accepted practice since there's no way for the client to know that a resource on the server was updated so the solution is specific code on the client that knows when to refresh it. Even if we were to return some special guid, it would probably be done as the result of `await avatarCreatePromise;` hence doing the same thing.

Reply:
So I tried doing what you suggested, 
```
await avatarCreatePromise;
setUser({...user});
```
However I think this is not working because nothing on the user has changed, so it is not re-rendered.
I'm going to try to return some kind of guid for the avatar...

Reply:
Here's what I came up with. It doesn't feel particularly elegant but it's re-rendering after uploading the avatar successfully.
https://github.com/TrevorSundberg/madeitforfun/pull/10/commits/d6591c6267ff4da0c74719b424e6a73742e806ea